### PR TITLE
API-1508: Log origin of each events

### PR DESCRIPTION
**Pull request by tomglvng**
EE: https://github.com/akeneo/pim-enterprise-dev/pull/10593

Origins are (CE):
 - UI
 - API
 - Mass action
 - Import

TODO:
 - [x] test create product from ui
 - [x] test remove product from ui
 - [x] test update product from ui
 - [x] test PM from ui
 - [x] test PM from API
 - [x] test Product from API
 - [x] test Import
 - [x] test Mass actions

### CODE REVIEW

File: src/Akeneo/Platform/Component/EventQueue/Event.php [hunk 1]
```diff
@@ -23,6 +24,8 @@ public function __construct(
         int $timestamp = null,
         string $uuid = null
     ) {
+        Assert::keyExists($data, 'origin');
```
Comment (tomglvng) [inline src/Akeneo/Platform/Component/EventQueue/Event.php:27]: TODO: Quid of existent events (without 'origin') ?

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductController.php [hunk 2]
```diff
@@ -351,7 +352,7 @@ public function removeAttributeAction(Request $request, $id, $attributeId)
                 $product->removeValue($value);
             }
         }
-        $this->productSaver->save($product);
+        $this->productSaver->save($product, ['origin' => 'UI']);
```
Comment (tseho) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductController.php:355]:
```suggestion
        $this->productSaver->save($product, ['origin' => ContextOrigin::UI]);
```

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductController.php [hunk 3]
```diff
@@ -384,7 +385,7 @@ public function convertToSimpleProductAction(Request $request, int $id): Respons
 
         $violations = $this->validator->validate($product);
         if (0 === $violations->count()) {
-            $this->productSaver->save($product);
+            $this->productSaver->save($product, ['origin' => 'UI']);
```
Comment (tseho) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductController.php:388]:
```suggestion
            $this->productSaver->save($product, ['origin' => ContextOrigin::UI]);
```

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductModelController.php [hunk 4]
```diff
@@ -369,7 +370,7 @@ public function removeAction(Request $request, $id): Response
         }
 
         $productModel = $this->findProductModelOr404($id);
-        $this->productModelRemover->remove($productModel);
+        $this->productModelRemover->remove($productModel, ['origin' => 'UI']);
```
Comment (tseho) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductModelController.php:373]:
```suggestion
        $this->productModelRemover->remove($productModel, ['origin' => ContextOrigin::UI]);
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/Job/JobParameters/DefaultValueProvider/ProductAndProductModelMassDelete.php [hunk 5]
```diff
@@ -15,12 +17,8 @@
  */
 class ProductAndProductModelMassDelete implements DefaultValuesProviderInterface
 {
-    /** @var string[] */
-    protected $supportedJobNames;
+    protected array $supportedJobNames;
```
Comment (tseho) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/Job/JobParameters/DefaultValueProvider/ProductAndProductModelMassDelete.php:20]: FYI, by removing the comment, you are removing the type of the items inside the array.
You should keep the comment also.

File: src/Akeneo/Platform/Component/EventQueue/Event.php (hunk 1 above)
Comment (tseho) [reply src/Akeneo/Platform/Component/EventQueue/Event.php:27]: I would remove the assert.

File: tests/back/Pim/Enrichment/Specification/Bundle/Controller/InternalApi/ProductControllerSpec.php [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+use Akeneo\UserManagement\Bundle\Context\UserContext;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Validator\ConstraintViolationList;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class ProductControllerSpec extends ObjectBehavior
```
Comment (tseho) [inline tests/back/Pim/Enrichment/Specification/Bundle/Controller/InternalApi/ProductControllerSpec.php:34]: Are you sure about a spec on this controller ?
Do the spec are relevant on controllers ?
This is also slightly out of scope.

**Review by tseho — APPROVED** [review]
GTM after fixes

**Pull request closed without merging**